Helm: Quote tenantId value in logsInstance

agrahamlincoln opened the pull request:
**What this PR does / why we need it**: Quotes the value of `tenantId` for the cases where tenantId is made up of only numerical values (i.e. if tenantId is `0`)

the motivation for this is that we have long used the tenant id `0` in our loki instance for some stuff. We probably should fix that, but it's going to be easier (at least for us) to patch this than to rename this tenant or migrate to a new tenant.

**Which issue(s) this PR fixes**:
N/A - If you'd like me to file an issue to track this there as well, let me know and i'll write one up. i figured this fix was simple enough

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/loki?pullRequest=9486) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/loki?pullRequest=9486) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/loki?pullRequest=9486) it.</sub>

agrahamlincoln commented:
I do not believe this fix is large enough to need to be added to CHANGELOG.md - feel free to disagree with me and I can figure out how to add it there :)

agrahamlincoln commented:
force pushed to fix casing from lowercase to `Sentence case` in the commit message

agrahamlincoln commented:
the drone test failed and i'm unsure how to fix it.

```
bash: ./cmd/loki/loki: Text file busy
make: *** [Makefile:771: validate-example-configs] Error 1
```

looks completely unrelated to my change, at a first glance to me this even looks like it could be a flakey test and a re-run might succeed.

MichelHollands commented:
@agrahamlincoln Can you add a changelog entry and update the Helm chart version? An example can be found [here](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213). Also please rebase.

agrahamlincoln commented:
Thank you for providing an example on how to correctly bump the version! I've updated my change to incorporate the changelog update. I force pushed to combine these changes with my existing ones and to put a changelog fix commit first before making my addition to the changelog.

When updating the changelog I noticed there was a changelog entry not attributed to a release: https://github.com/grafana/loki/commit/0973297ac3d9bce1679741e15ba44e2bfcdc1193

I fixed this in my updated changes

MichelHollands commented:
@agrahamlincoln This looks fine now. But it seem another commit got in before. Can you fix the merge conflicts? I'll merge this next once that is done.

agrahamlincoln commented:
all fixed @MichelHollands ! looks like someone incorrectly attributed that dangling changelog commit to 5.5.4 - however, the commit was first released in 5.5.2 so I fixed that. it shouldn't make a big difference but it was easy to fix.
